Files.Launcher/Program.cs

```diff
@@ -1,85 +1,319 @@
+            connection = new AppServiceConnection();
+            connection.AppServiceName = "FilesInteropService";
+            connection.PackageFamilyName = Package.Current.Id.FamilyName;
+            connection.RequestReceived += Connection_RequestReceived;
+            connection.ServiceClosed += Connection_ServiceClosed;
+
+            AppServiceConnectionStatus status = await connection.OpenAsync();
+            if (status != AppServiceConnectionStatus.Success)
+            {
+                // TODO: error handling
+                connection.Dispose();
+                connection = null;
+            }
+        }
+
+        private static async void Connection_RequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            // Get a deferral because we use an awaitable API below to respond to the message
+            // and we don't want this call to get cancelled while we are waiting.
+            var messageDeferral = args.GetDeferral();
+
+            if (args.Request.Message == null)
+            {
+                messageDeferral.Complete();
+                return;
+            }
+
+            try
+            {
+                if (args.Request.Message.ContainsKey("Arguments"))
+                {
+                    // This replaces launching the fulltrust process with arguments
+                    // Instead a single instance of the process is running
+                    // Requests from UWP app are sent via AppService connection
+                    var arguments = (string)args.Request.Message["Arguments"];
+                    var localSettings = Windows.Storage.ApplicationData.Current.LocalSettings;
+
+                    if (arguments == "Terminate")
+                    {
+                        // Exit fulltrust process (UWP is closed or suspended)
+                        appServiceExit.Set();
+                        messageDeferral.Complete();
+                    }
+                    else if (arguments == "RecycleBin")
+                    {
+                        var action = (string)args.Request.Message["action"];
+                        if (action == "Empty")
+                        {
+                            // Shell function to empty recyclebin
+                            SHEmptyRecycleBin(IntPtr.Zero, null, RecycleFlags.SHERB_NOCONFIRMATION | RecycleFlags.SHERB_NOPROGRESSUI);
+                        }
+                        else if (action == "Enumerate")
+                        {
+                            // Enumerate recyclebin contents and send response to UWP
+                            var response_enum = new ValueSet();
+                            Shell32.FolderItems items = Recycler.Items();
+                            var file_list = new List<ShellFileItem>();
+                            for (int i = 0; i < items.Count; i++)
+                            {
+                                Shell32.FolderItem FI = items.Item(i);
```

Comment:
If this vars are inside loop, so how to use camelCase?

Reply:
You forget to change this virus names to camelCase. (Currently they are used CamelCase)